base_agent/loco_mc_agent.py

```diff
@@ -218,24 +218,23 @@ def step(self):
         self.maybe_dump_memory_to_dashboard()
 
     def task_step(self, sleep_time=0.25):
-        # Clean finished tasks
-        while (
-            self.memory.task_stack_peek() and self.memory.task_stack_peek().task.check_finished()
-        ):
-            self.memory.task_stack_pop()
-
-        # If nothing to do, wait a moment
-        if self.memory.task_stack_peek() is None:
+        query = {"base_table": "Tasks", "base_range": {"minprio": -0.5, "maxpaused": 0.5}}
```

Comment:
this looks significantly more involved than before. Please add documentation for this. What do the two queries represent? - ` query = {"base_table": "Tasks", "base_range": {"minprio": -0.5, "maxpaused": 0.5}}` and `query = {"base_table": "Tasks", "base_range": {"minrunning": 0.5}}`

Reply:
this is a good example of the SQL++ discussion from a couple of days ago.  This query in SQL would be self documenting; but then would be raw SQL in the agent (and so would break if the backend db was switched to some other key-value store etc.   If we had our own dialect of SQL and parsed it (and made our own interface), all the queries would be clearer...  fwiw its "SELECT * FROM Tasks WHERE prio > -.5 and paused < .5", so select all tasks where the prio allows the on_condition to be checked and where the task has not been explicitly paused by the player